lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +38,50 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._scan_dataset()
+        self._validate_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if len(list(self.path_to_raw_txt_data.iterdir())) == 0:
+            raise EmptyDirectoryError
+
+        texts = [text for text in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
+        text_sizes = [os.path.getsize(text) for text in texts]
+
+        metas = [meta for meta in self.path_to_raw_txt_data.glob(r'*_meta.json')]
+        meta_sizes = [os.path.getsize(text) for text in metas]
+
+        if len(texts) != len(metas):
+            raise InconsistentDatasetError
+
+        if 0 in text_sizes or 0 in meta_sizes:
+            raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry
         """
+        for text in self.path_to_raw_txt_data.glob('*.txt'):
+            for index in range(999):
```

Comment:
this is very creative!
however, try to find a more elegant way of determining the index of the article. consider extracting it from the fime name or from the article instance

Reply:
fixed